test/test_integration_credentials_netrc.py

```diff
@@ -0,0 +1,413 @@
+        utils.setup_cli(self)
+        self.invoke_cli(self.cli_auth_params,
+                        ['--insecure', 'init', '--url', self.client_params.hostname, '--account',
+                         self.client_params.account])
+
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['--insecure', 'login', '-i', 'admin', '-p', self.client_params.env_api_key])
+        self.assertIn('Successfully logged in to Conjur', output)
+
+    '''
+    Validates a user can log in with a password, instead of their API key
+    To do this, we perform the following:
+    '''
+
+    @integration_test()
+    @patch('builtins.input', return_value='yes')
+    def test_https_netrc_is_created_with_all_parameters_given(self, mock_input,keystore_disable_mock):
+        utils.setup_cli(self)
+        self.invoke_cli(self.cli_auth_params,
+                        ['login', '-i', 'admin', '-p', self.client_params.env_api_key])
+
+        # Load in new user
+        self.invoke_cli(self.cli_auth_params,
+                        ['policy', 'replace', '-b', 'root', '-f',
+                         self.environment.path_provider.get_policy_path('conjur')])
+
+        # Rotate the new user's API key
+        user_api_key = self.invoke_cli(self.cli_auth_params,
+                                       ['user', 'rotate-api-key', '-i', 'someuser'])
+        extract_api_key_from_message = user_api_key.split(":")[1].strip()
+
+        # Login to change personal password
+        self.invoke_cli(self.cli_auth_params,
+                        ['login', '-i', 'someuser', '-p', extract_api_key_from_message])
+
+        # Creates a password that meets Conjur password complexity standards
+        password = string.hexdigits + "$!@"
+        self.invoke_cli(self.cli_auth_params,
+                        ['user', 'change-password', '-p', password])
+
+        self.invoke_cli(self.cli_auth_params,
+                        ['logout'])
+
+        # Attempt to login with newly created password
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['login', '-i', 'someuser', '-p', password])
+
+        self.assertIn("Successfully logged in to Conjur", output.strip())
+        utils.get_credentials()
+        self.validate_netrc(f"{self.client_params.hostname}", "someuser", extract_api_key_from_message)
+
+    '''
+    Validates a wrong username will raise Unauthorized error
+    '''
+
+    @integration_test()
+    @patch('builtins.input', return_value='somebaduser')
+    def test_https_netrc_raises_error_with_wrong_user(self, mock_pass,keystore_disable_mock):
+        utils.setup_cli(self)
+        with patch('getpass.getpass', return_value=self.client_params.env_api_key):
+            output = self.invoke_cli(self.cli_auth_params,
+                                     ['login'], exit_code=1)
+
+            self.assertRegex(output, "Reason: 401")
+
```

Comment:
`assert not os.path.exists(DEFAULT_NETRC_FILE)`

is missing from this test

Reply:
why should the DEFAULT_NETRC_FILE be deleted here? we just empty the file no?